src/box/lua/schema.lua

```diff
@@ -3593,30 +3593,84 @@ local function role_check_grant_revoke_of_sys_priv(priv)
     local uid = role_resolve(user_name)
     if uid == nil then
         box.error(box.error.NO_SUCH_ROLE, user_name)
     end
     role_check_grant_revoke_of_sys_priv(...)
     return revoke(uid, user_name, ...)
 end
-box.schema.role.info = function(role_name)
+
+local function box_schema_role_info_do(role_name)
     local rid = role_resolve(role_name)
     if rid == nil then
         box.error(box.error.NO_SUCH_ROLE, role_name)
     end
     return info(rid)
 end
 
+-- Wrap a function into transaction if none is active.
+local function atomic_wrapper(func)
+    return function(...)
+        -- No reason to start a transaction of one is active already.
+        if box.is_in_txn() then
+            return func(...)
+        else
+            return box.atomic(func, ...)
+        end
+    end
+end
+
+box.schema.space.create = atomic_wrapper(box_schema_space_create_do)
+box.schema.space.format = atomic_wrapper(box_schema_space_format_do)
+box.schema.space.upgrade = box_schema_space_upgrade_do
+box.schema.space.drop = atomic_wrapper(box_schema_space_drop_do)
+box.schema.space.rename = atomic_wrapper(box_schema_space_rename_do)
+box.schema.space.alter = atomic_wrapper(box_schema_space_alter_do)
+
+box.schema.index.create = atomic_wrapper(box_schema_index_create_do)
+box.schema.index.drop = atomic_wrapper(box_schema_index_drop_do)
+box.schema.index.rename = atomic_wrapper(box_schema_index_rename_do)
+box.schema.index.alter = box_schema_index_alter_do
+
+box.schema.sequence.create = atomic_wrapper(box_schema_sequence_create_do)
+box.schema.sequence.alter = atomic_wrapper(box_schema_sequence_alter_do)
+box.schema.sequence.drop = atomic_wrapper(box_schema_sequence_drop_do)
+
+box.schema.func.create = atomic_wrapper(box_schema_func_create_do)
+box.schema.func.drop = atomic_wrapper(box_schema_func_drop_do)
+box.schema.func.exists = atomic_wrapper(box_schema_func_exists_do)
+
+box.schema.user.password = atomic_wrapper(box_schema_user_password_do)
+box.schema.user.passwd = atomic_wrapper(box_schema_user_passwd_do)
+box.schema.user.create = atomic_wrapper(box_schema_user_create_do)
+box.schema.user.exists = atomic_wrapper(box_schema_user_exists_do)
+box.schema.user.grant = atomic_wrapper(box_schema_user_grant_do)
+box.schema.user.revoke = atomic_wrapper(box_schema_user_revoke_do)
+box.schema.user.enable = atomic_wrapper(box_schema_user_enable_do)
+box.schema.user.disable = atomic_wrapper(box_schema_user_disable_do)
+box.schema.user.drop = atomic_wrapper(box_schema_user_drop_do)
+box.schema.user.info = atomic_wrapper(box_schema_user_info_do)
+
+box.schema.role.exists = atomic_wrapper(box_schema_role_exists_do)
+box.schema.role.create = atomic_wrapper(box_schema_role_create_do)
+box.schema.role.drop = atomic_wrapper(box_schema_role_drop_do)
+box.schema.role.grant = atomic_wrapper(box_schema_role_grant_do)
+box.schema.role.revoke = atomic_wrapper(box_schema_role_revoke_do)
+box.schema.role.info = atomic_wrapper(box_schema_role_info_do)
```

Comment:
`info` and `exists` methods are read-only so they don't have any side-effects hence they don't need the atomic wrapper, do they?

Reply:
See the [answer](https://github.com/tarantool/tarantool/pull/8751#discussion_r1317401750) to the previous comment.